bugshaker/src/main/java/com/github/stkent/bugshaker/flow/email/FeedbackEmailFlowManager.java

```diff
@@ -30,221 +35,270 @@
+
+	public FeedbackEmailFlowManager(ScreenshotProvider screenshotProvider, Application application) {
+		this.screenshotProvider = ScreenshotUtil.getScreenshotProvider(application);
+		activityReferenceManager = new ActivityReferenceManager();
+
+	}
+
+	private final OnClickListener screenshotListener = new OnClickListener() {
+		@Override
+		public void onClick(DialogInterface dialogInterface, int i) {
+
+			final Activity activity = activityReferenceManager.getValidatedActivity();
+			final Context context = activity.getBaseContext();
+			final Toaster toaster = new Toaster(activity);
+			final Logger logger = new Logger(true);
+			if (activity == null) {
+				return;
+			}
+
+			if (shouldAttemptToCaptureScreenshot(activity)) {
+				if (EmailCapabilitiesProvider.canSendEmailsWithAttachments(
+					activityReferenceManager.getValidatedActivity().getPackageManager())) {
+					screenshotProvider.getScreenshotUri(activity)
+						.single()
+						.observeOn(AndroidSchedulers.mainThread())
+						.subscribeOn(AndroidSchedulers.mainThread())
+						.subscribe(new Subscriber<Uri>() {
+							@Override
+							public void onCompleted() {
+								// This method intentionally left blank.
+							}
+
+							@Override
+							public void onError(final Throwable e) {
+								final String errorString = "Screenshot capture failed";
+								toaster.toast(errorString);
+								logger.e(errorString);
+								logger.printStackTrace(e);
+							}
+
+							@Override
+							public void onNext(final Uri uri) {
+								startActivity(context);
+							}
+						});
+				}
+			}
+			else {
+				final String warningString = "Window is secured; no screenshot taken";
+				toaster.toast(warningString);
+				logger.d(warningString);
+			}
+
+		}
+
+	};
+
+	private final OnClickListener reportBugClickListener = new OnClickListener() {
+		@Override
+		public void onClick(final DialogInterface dialog, final int which) {
+			final Activity activity = activityReferenceManager.getValidatedActivity();
+			final Context context = activity.getBaseContext();
+			final Toaster toaster = new Toaster(activity);
+			final Logger logger = new Logger(true);
+			if (activity == null) {
+				return;
+			}
+
+			if (shouldAttemptToCaptureScreenshot(activity)) {
+				if (EmailCapabilitiesProvider.canSendEmailsWithAttachments
+					(activityReferenceManager.getValidatedActivity().getPackageManager())) {
+					screenshotProvider.getScreenshotUri(activity)
+						.single()
+						.observeOn(AndroidSchedulers.mainThread())
+						.subscribeOn(AndroidSchedulers.mainThread())
+						.subscribe(new Subscriber<Uri>() {
+							@Override
+							public void onCompleted() {
+								// This method intentionally left blank.
+							}
+
+							@Override
+							public void onError(final Throwable e) {
+								final String errorString = "Screenshot capture failed";
+								toaster.toast(errorString);
+								logger.e(errorString);
+
+								logger.printStackTrace(e);
```

Comment:
we removed logger.printStackTrace(e), were we supposed to remove anything else though?